client/app/authenticators/zap-api-authenticator.js

```diff
@@ -1,79 +1,32 @@
 import { InvalidError } from '@ember-data/adapter/error';
 import ENV from '../config/environment';
 
-// lifted from https://github.com/simplabs/ember-simple-auth/blob/master/addon/mixins/oauth2-implicit-grant-callback-route-mixin.js#L6
-// parses a window hash and grabs the access token
-function _parseResponse(locationHash) {
-  const params = {};
-  const query = locationHash.substring(locationHash.indexOf('?'));
-  const regex = /([^#?&=]+)=([^&]*)/g;
-  let match;
+export default class ZAPAuthenticator extends OAuth2ImplicitGrantAuthenticator {
+  async authenticate(...args) {
+    let accessToken;
 
-  // decode all parameter pairs
-  while ((match = regex.exec(query)) !== null) { // eslint-disable-line
-    params[decodeURIComponent(match[1])] = decodeURIComponent(match[2]);
-  }
-
-  return params;
-}
-
-export default class ZAPAuthenticator extends BaseAuthenticator {
-  @service
-  store
-
-  // Authentication is implicit with an httponly cookie holding
-  // a ZAP Token. This request will fail if that cookie doesn't exist
-  // https://github.com/simplabs/ember-simple-auth/blob/master/guides/managing-current-user.md#using-a-dedicated-endpoint
-  async _fetchContact() {
-    const { id, emailaddress1, contactid } = await this.store.queryRecord('contact', { me: true });
-
-    return { id, emailaddress1, contactid };
-  }
-
-  async authenticate() {
-    // Extract NYCID Token from URL
-    const { access_token } = _parseResponse(window.location.hash);
+    try {
+      const { access_token } = await super.authenticate(...args);
 
-    if (!access_token) {
-      throw new InvalidError([{
-        response: {
-          code: 'NO_TOKEN_PRESENT',
-        },
-      }]);
+      accessToken = access_token;
+    } catch (e) {
+      throw new InvalidError([{ detail: e.toString() }]);
     }
 
     // Pass the NYCIDToken to backend /login endpoint.
-    // The backend's response will contain a Cookie with a ZAP token.
-    // This Cookie is automatically stored away and used in future requests to the
-    // backend.
-    const response = await fetch(`${ENV.host}/login?accessToken=${access_token}`, {
-      mode: 'same-origin',
-      credentials: 'include',
-    });
+    // Server should provide an access_token used for signing
+    // requests using the Authorization header
+    const response = await fetch(`${ENV.host}/login?accessToken=${accessToken}`);
+    const body = await response.json();
 
-    if (!response.ok) throw await response.json();
+    if (!response.ok) throw body;
 
-    // Since all requests to the API are now authenticated,
-    // formally request a User through Ember Data and return it.
-    // Returning the user data here places it inside
-    // the session data's "authenticated" property and marks
+    // Since all requests to the API are now authenticated.
+    // Returning the session data's "authenticated" property and marks
```

Comment:
I don't understand this comment here. How are you returning the session data's authenticated property here?

Reply:
ah yeah i'm referring to this: 
![image](https://user-images.githubusercontent.com/5004319/88973132-a725a680-d284-11ea-98d0-1277b0b63c7c.png)

http://ember-simple-auth.com/api/classes/SessionService.html#method_authenticate

Reply:
Oh I see, here it's returning _to_ the session data's authenticated property